File: src/matlab/+jterator/+api/+io/read_input_values.m
%% Reading initial values from "handles".

function values = read_input_values(handles)

    values = struct();
    keys = fieldnames(handles.input_keys);
    for i = 1:length(keys)
        key = keys{i};
        values.(key) = handles.input_keys.(key).value;
        fprintf('jt -- %s: value ''%s'': "%s"\n', ...
            mfilename, key, values.(key))
    end
    
end

File: src/r/jterator/R/read_input_values.R
read_input_values <-
function(handles) {

    mfilename <- "read_input_values"

    values <- list()
    for (key in names(handles$input_keys)) {
      values[key] <- handles$input_keys[[key]]$value
      cat(sprintf("jt -- %s: value '%s': \"%s\"\n",
                  mfilename, key, values[key]))
    }

    return(values)
}
